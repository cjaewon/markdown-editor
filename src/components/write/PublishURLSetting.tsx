import * as React from 'react';
import styled from 'styled-components';
import PublishSection from './PublishSection';
import palette from '../../lib/styles/palette';

const PublishURLSettingBlock = styled(PublishSection)``;
const URLTextBlock = styled.div`
  display: flex;
  background: white;
  box-shadow: 0 0 4px 0 rgba(0, 0, 0, 0.03);
  padding: 0.5rem 0.875rem;
  line-height: 1.5;
  .username {
    color: ${palette.gray6};
  }
`;
const Input = styled.input`
  font-size: 1rem;
  background: none;
  outline: none;
  flex: 1;
  border: none;
  padding: 0;
  line-height: 1.5;
  color: ${palette.gray8};
`;

export interface PublishURLSettingProps {
  username: string;
  urlSlug: string;
  onChangeUrlSlug: (urlSlug: string) => void;
}

const PublishURLSetting: React.FC<PublishURLSettingProps> = ({
  username,
  urlSlug,
  onChangeUrlSlug,
}) => {
  const onChange = React.useCallback(
    (e: React.ChangeEvent<HTMLInputElement>) => onChangeUrlSlug(e.target.value),
    [onChangeUrlSlug],
  );
  return (
    <PublishURLSettingBlock title="URL 설정">
      <URLTextBlock>
        <div className="username">/@{username}/</div>
        <Input value={urlSlug} onChange={onChange} />
      </URLTextBlock>
    </PublishURLSettingBlock>
  );
};

export default PublishURLSetting;
