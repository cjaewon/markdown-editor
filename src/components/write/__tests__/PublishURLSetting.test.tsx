import * as React from 'react';
import { render, fireEvent } from '@testing-library/react';
import PublishURLSetting, {
  PublishURLSettingProps,
} from '../PublishURLSetting';

describe('PublishURLSetting', () => {
  const setup = (props: Partial<PublishURLSettingProps> = {}) => {
    const initialProps: PublishURLSettingProps = {
      username: 'velog',
      urlSlug: 'url-slug',
      onChangeUrlSlug: () => {},
    };
    const utils = render(<PublishURLSetting {...initialProps} {...props} />);
    return {
      ...utils,
    };
  };
  it('renders properly', () => {
    setup();
  });
  it('matches snapshot', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
  it('shows username props', () => {
    const utils = setup();
    utils.getByText('/@velog/');
  });
  it('urlSlug is working properly', () => {
    const onChangeUrlSlug = jest.fn();
    const utils = setup({
      onChangeUrlSlug,
    });
    const input = utils.getByDisplayValue('url-slug');
    fireEvent.change(input, { target: { value: 'hello-world' } });
    expect(onChangeUrlSlug).toBeCalledWith('hello-world');
  });
});
