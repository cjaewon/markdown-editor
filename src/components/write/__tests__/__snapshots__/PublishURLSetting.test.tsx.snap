// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublishURLSetting matches snapshot 1`] = `
<div>
  <section
    class="sc-bdVaJa ezdyV sc-bwzfXH uTxCW"
  >
    <h3>
      URL 설정
    </h3>
    <div
      class="contents"
    >
      <div
        class="sc-htpNat hgJOta"
      >
        <div
          class="username"
        >
          /@
          velog
          /
        </div>
        <input
          class="sc-bxivhb knHhGg"
          value="url-slug"
        />
      </div>
    </div>
  </section>
</div>
`;
